### ISSUE TITLE
[Bug]  panic: http: wrote more than the declared Content-Length

### ISSUE THREAD

**Issue by hantmac** [gin-contrib/timeout#1]
After timeout, the original goroutine still is running, and will cause panic finally.

The panic info is:
```
[GIN-debug] [WARNING] Headers were already written. Wanted to override status code 408 with 200
panic: http: wrote more than the declared Content-Length

```

![image](https://user-images.githubusercontent.com/7600925/91701166-79948b80-eba9-11ea-98c2-155e9583764d.png)

**Comment by fnsne** [comment]
I found the solution in the other timeout implement.
https://github.com/vearne/gin-timeout
By injecting other Writer to the wrapped function, and using the select case to chose what to write.

**Comment by hantmac** [comment]
> I found the solution in the other timeout implement.
> https://github.com/vearne/gin-timeout
> By injecting other Writer to the wrapped function, and using the select case to chose what to write.

I saw it, but I think his code is so complicated 🤔.

**Comment by appleboy** [comment]
Let me think about how to resolve the issue.

**Comment by jdeal-mediamath** [comment]
Any progress on fixing this?

**Comment by jdeal-mediamath** [comment]
Found a workaround until this is fixed.

Just check if the context is aborted anywhere else that attempts to write. Not ideal, but it worked for me.

**Comment by justlazydog** [comment]
Is it possible that gin's repo provides a func to copy or clone c.Writer.

**Comment by appleboy** [comment]
Fix by #2
